### PR TITLE
Check android sdk sha256 sum on Dockerfile

### PR DESCRIPTION
**Pull request by spk**
// FREEBIE

### First time contributor checklist
- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description

Check android sdk sha256 sum on Dockerfile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,15 +11,17 @@ RUN dpkg --add-architecture i386 && \
...
 ENV ANDROID_SDK_FILENAME android-sdk_r24.4.1-linux.tgz
+ENV ANDROID_SDK_SHA256 e16917ad685c1563ccbc5dd782930ee1a700a1b6a6fd3e44b83ac694650435e9
 ENV ANDROID_SDK_URL https://dl.google.com/android/${ANDROID_SDK_FILENAME}
 ENV ANDROID_API_LEVELS android-25
 ENV ANDROID_BUILD_TOOLS_VERSION 23.0.3
 ENV ANDROID_HOME /usr/local/android-sdk-linux
 ENV PATH ${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/platform-tools
 RUN cd /usr/local/ && \
     wget -q ${ANDROID_SDK_URL} && \
+    echo "${ANDROID_SDK_SHA256} ${ANDROID_SDK_FILENAME}" | sha256sum -c - && \
```
Comment (Stuie) [inline Dockerfile]: Excuse my ignorance of Docker, but does adding this line require that users have the `sha256sum` binary in their `$PATH`?

Comment (2-4601) [inline Dockerfile]: Nope. It's only required that the base image (in this case `ubuntu:14.04.03`) has it.

Comment (Stuie) [inline Dockerfile]: It seems the input [requires two spaces](http://unix.stackexchange.com/a/139892) for certain versions (e.g. 8.13) of the sha256sum binary. Can we guarantee that the base image does not contain that version?

Looks good to me once we know that.

Comment (2-4601) [inline Dockerfile]:
```
$ docker run --rm ubuntu:14.04.3 sha256sum --version
sha256sum (GNU coreutils) 8.21
```

File: Dockerfile [hunk 2]
```diff
@@ -10,15 +10,17 @@ RUN dpkg --add-architecture i386 && \
...
 ENV ANDROID_SDK_FILENAME android-sdk_r24.4.1-linux.tgz
+ENV ANDROID_SDK_SHA256 e16917ad685c1563ccbc5dd782930ee1a700a1b6a6fd3e44b83ac694650435e9
 ENV ANDROID_SDK_URL https://dl.google.com/android/${ANDROID_SDK_FILENAME}
 ENV ANDROID_API_LEVELS android-27
 ENV ANDROID_BUILD_TOOLS_VERSION 27.0.1
 ENV ANDROID_HOME /usr/local/android-sdk-linux
 ENV PATH ${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/platform-tools
 RUN cd /usr/local/ && \
     wget -q ${ANDROID_SDK_URL} && \
+    echo "${ANDROID_SDK_SHA256} ${ANDROID_SDK_FILENAME}" | sha256sum -c - && \
```
Comment (tcely) [inline Dockerfile]:
```suggestion
    printf -- '%s  %s\n' "${ANDROID_SDK_SHA256}" "${ANDROID_SDK_FILENAME}" | sha256sum -c - && \
```

Comment (spk) [reply Dockerfile]: Applied in ba01ddd1c

**Comment by spk** [comment]
Project moved to https://github.com/signalapp/Signal-Android/tree/master/reproducible-builds

**Pull request closed without merging**